src/Equinor.ProCoSys.IPO.Query/GetInvitations/GetInvitationsQueryBase.cs

```diff
@@ -0,0 +1,202 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Equinor.ProCoSys.IPO.Domain;
+using Equinor.ProCoSys.IPO.Domain.AggregateModels.InvitationAggregate;
+using Microsoft.EntityFrameworkCore;
+
+namespace Equinor.ProCoSys.IPO.Query.GetInvitations
+{
+    public abstract class GetInvitationsQueryBase
+    {
+        protected IQueryable<InvitationDto> CreateQueryableWithFilter(IReadOnlyContext context, string projectName, Filter filter, DateTime utcNow)
+        {
+            var startOfThisWeekUtc = DateTime.MinValue;
+            var startOfNextWeekUtc = DateTime.MinValue;
+            var startOfTwoWeeksUtc = DateTime.MinValue;
+            if (filter.PunchOutDates.Any())
+            {
+                startOfThisWeekUtc = utcNow.StartOfWeek();
+                startOfNextWeekUtc = startOfThisWeekUtc.AddWeeks(1);
+                startOfTwoWeeksUtc = startOfThisWeekUtc.AddWeeks(2);
+            }
+
+            var ipoIdStartWith = GetIpoIdStartWith(filter.IpoIdStartsWith);
+ 
+            var queryable = from invitation in context.QuerySet<Invitation>()
+                where invitation.ProjectName == projectName &&
+                      (!filter.PunchOutDates.Any() ||
+                           (filter.PunchOutDates.Contains(PunchOutDateFilterType.Overdue) && invitation.StartTimeUtc < startOfThisWeekUtc) ||
```

Comment:
If the IPO round is invited to happen on Tuesday at 14:30, is it overdue already when the particular week starts?
Isn't overdue when invitation.StartTimeUtc > utcNow?

Reply:
Yes, true, will fix this!